index.js

```diff
@@ -24,12 +28,45 @@ module.exports = async function hubdown (markdownString, opts = {}) {
   let data = {}
   let content = markdownString
 
+  // check the cache for preprocessed markdown
+  if (opts.cache) {
+    let existing = false
+    try {
+      existing = await opts.cache.get(hash)
+    } catch (err) {
+      if (!err.notFound) console.error(err)
+    }
+    if (existing) return existing
+  }
+
   if (opts.frontmatter) {
     const parsed = grayMatter(markdownString)
     data = parsed.data
     content = parsed.content
   }
 
   const md = await pify(renderer.process)(content)
-  return Object.assign(data, {content: md.contents})
+  Object.assign(data, {content: md.contents})
+
+  // save processed markdown in cache
+  if (opts.cache) await opts.cache.put(hash, data)
+
+  return data
+}
+
+// create a unique hash from the given input (markdown + options object)
+function makeHash (markdownString, opts) {
+  // copy existing opts object to avoid mutation
+  const hashableOpts = Object.assign({}, opts)
+
+  // ignore `cache` prop when creating hash
+  delete hashableOpts.cache
+
+  // object keys are sorted to ensure {a:1, b:2} has the same hash as {b:2, a:1}
+  // empty object should become an empty string, not {}
+  const optsString = Object.keys(hashableOpts).length
+    ? JSON.stringify(sortObject(hashableOpts))
+    : ''
```

Comment:
I had the same use case in the colors hash branch and used `stable-json-stringify` which seems to be more widely used than `sort-object`. 

What would you think about using that here too?

Reply:
I didn't know about `json-stable-stringify`. Swapped it out. Thanks for the tip. This is ready for review again.